### DISCUSSION TITLE
How to add a new option to the GraphicsLayoutWidget of right-click menu?

### DISCUSSION THREAD

**Discussion by ningwana** [pyqtgraph/pyqtgraph#2666] [Q&A]
Now it has these options, how do I add other options(eg:Test options)? I was looking for a similar problem on stackoverflow, but it didn't work out.

-----
- View All
- X axis
- Y axis
- Mouse Mode
- Plot Options
- Export
- **Test options**

-----
If anyone responds to the message, I would appreciate it.
If I don't describe it in detail, please leave a message.

**Comment by j9ac9k** [comment]
I don't think the library has a particularly great way of adding to the existing set of options; probably doable tho.

Here is the relevant bit of code: https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/graphicsItems/PlotItem/PlotItem.py#L189-L205

Any custom action you would need to add to the `PlotItem.ctrlMenu` `QMenu` and go from there.

**Reply by ningwana** [reply]
Thank you for your answer.
What I want to achieve is without affecting the original window, and in the pop-up of another window to display the data.
It's like: Can I get everything from an GraphicsLayoutWidget by double clicking and copy it to another GraphicsLayoutWidget?
Then when a class instantiates myplot, you can do it by double-clicking.

This is an example
```python
from PySide2.QtWidgets import *

from pyqtgraph import GraphicsLayoutWidget


class MyPlot(GraphicsLayoutWidget):

    def __init__(self, parent=None):
        self.parent = parent
        self.window = None
        super(MyPlot, self).__init__(parent)

    def mouseDoubleClickEvent(self, ev):
        super(MyPlot, self).mouseDoubleClickEvent(ev)
        print("==================")

        self.window = GraphicsLayoutWidget(show=True)
        #  Can I get everything from an GraphicsLayoutWidget by double clicking and copy it to another 
        self.p2 = self.window.addPlot(0, 0)
        self.p2.plot(self.parent.x, self.parent.y)

        self.window.show()


class MyWindow(QWidget):
    def __init__(self, parent=None):
        super(MyWindow, self).__init__(parent)
        self.resize(800, 600)
        self.x = range(0, 10)
        self.y = [2, 4, 6, 8, 10, 12, 14, 16, 18, 20]

        self.pw = MyPlot(self)
        self.p1 = self.pw.addPlot(0, 0)
        self.p1.plot(self.x, self.y)

        layout = QHBoxLayout()
        layout.addWidget(self.pw)
        self.setLayout(layout)


if __name__ == "__main__":
    import sys

    app = QApplication()
    main = MyWindow()
    main.show()
    sys.exit(app.exec_())


```

**Reply by ningwana** [reply]
In general, I have some GraphicsLayoutWidget components in my project. I want to double-click GraphicsLayoutWidget to pop up a new window without affecting the original function. This window has everything from the original window.
